galeracluster/source/installmariadbsrc.rst

```diff
@@ -5,61 +5,189 @@ Source Installation
+Preparing the Server
+---------------------------------
+.. _`installmariadb-prep-server`:
+
+When building from source code, ``make`` cannot manage or install dependencies for either Galera Cluster or the build process itself.  You need to install these packages first.
+
+- For Debian-based distributions of Linux, if MariaDB is available in your repositories, you can run the following command:
+
+  .. code-block:: console
+
+     # apt-get build-dep mariadb-server
+
+- For RPM-based distributions, instead run this command:
+
+  .. code-block:: console
+
+     # yum-builddep MariaDB-server
+
+In the event that neither command works for your system or that you use a different Linux distribution or FreeBSD, the following packages are required:
+
+- **MariaDB Database Server with wsrep API**: Git, CMake, GCC and GCC-C++, Automake, Autoconf, and Bison, as well as development releases of libaio and ncurses.
+
+- **Galera Replication Plugin**: SCons, as well as development releases of Boost, Check and OpenSSL.
+
+Check with the repositories for your distribution or system for the appropriate package names to use during installation.  Bear in mind that different systems may use different names and that some may require additional packages to run.  For instance, to run CMake on Fedora you need both ``cmake`` and ``cmake-fedora``.  
+
+     
 -----------------------------------------
-Build Dependencies
+Building MariaDB Galera Cluster
 -----------------------------------------
-.. _`Build Dependencies`:
+.. _`build-galera-mariadb`:
+
+The source code for MariaDB Galera Cluster is available through `GitHub <https://github.com>`_. Using Git you can download the source code to build MariaDB and the Galera Replicator Plugin locally on your system.
+
+#. Create a directory for the MariaDB database server source files.
+
+   .. code-block:: console
+
+      # mkdir mariadb
+
+#. From within that directory, initialize a Git repository.
+
+   .. code-block:: console
+
+      # cd mariadb
+      # git init
+
+#. For your local Git repository, add the web address for the MariaDB server on GitHub.
+
+   .. code-block:: console
+
+      # git remote add origin https://github.com/mariadb/server
 
-Before you begin building MariaDB Galera Cluster from source, you must first install the build dependencies on your server.  If your system uses a Debian-based distribution of Linux, run the following command:
+#. Fetch the source files from the origin server, (that is, from GitHub).
 
+   .. code-block:: console
+
+      # git fetch origin
+      
+#. Switch the repository to the branch that you want to build.
+
+
+   MariaDB version 10.1 is packaged with Galera Cluster, prior to 10.1, branches that include Galera Cluster have the ``-galera`` suffix.  The main branches are:
+
+   .. warning:: MariaDB version 10.1 is in beta.
+   
+   - 10.1
+   - 10.0-galera
+   - 5.5-galera
+
+   .. code-block:: console
+
+      # git checkout -b 10.0-galera origin/10.0-galera
```

Comment:
git checkout 10.0-galera is enough to cause the local branch to be created automatically.